File: sdks/java/src/main/java/com/blok/blok/types/NodeConfig.java
package com.blok.blok.types;

import com.google.gson.annotations.SerializedName;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

/**
 * Represents node-specific configuration from the Blok runner.
 */
public class NodeConfig {

    private String name;
    private String path;

    @SerializedName("type")
    private String type;

    private Map<String, Object> config;

    public NodeConfig() {
        this.config = new HashMap<>();
    }

    /**
     * Retrieves a string config value with a default.
     *
     * @param key        the config key
     * @param defaultVal the default value if key is missing or not a string
     * @return the config value or default
     */
    public String getConfigString(String key, String defaultVal) {
        if (config == null || key == null) {
            return defaultVal;
        }
        Object value = config.get(key);
        if (value instanceof String) {
            return (String) value;
        }
        return defaultVal;
    }

    /**
     * Retrieves an integer config value with a default.
     *
     * @param key        the config key
     * @param defaultVal the default value if key is missing or not numeric
     * @return the config value or default
     */
    public int getConfigInt(String key, int defaultVal) {
        if (config == null || key == null) {
            return defaultVal;
        }
        Object value = config.get(key);
        if (value instanceof Number) {
            return ((Number) value).intValue();
        }
        return defaultVal;
    }

    /**
     * Retrieves a boolean config value with a default.
     *
     * @param key        the config key
     * @param defaultVal the default value if key is missing or not a boolean
     * @return the config value or default
     */
    public boolean getConfigBool(String key, boolean defaultVal) {
        if (config == null || key == null) {
            return defaultVal;
        }
        Object value = config.get(key);
        if (value instanceof Boolean) {
            return (Boolean) value;
        }
        return defaultVal;
    }

    // Getters and setters

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Map<String, Object> getConfig() {
        return config;
    }

    public void setConfig(Map<String, Object> config) {
        this.config = config != null ? config : new HashMap<>();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        NodeConfig that = (NodeConfig) o;
        return Objects.equals(name, that.name) &&
                Objects.equals(path, that.path) &&
                Objects.equals(type, that.type) &&
                Objects.equals(config, that.config);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, path, type, config);
    }

    @Override
    public String toString() {
        return "NodeConfig{" +
                "name='" + name + '\'' +
                ", path='" + path + '\'' +
                ", type='" + type + '\'' +
                ", config=" + config +
                '}';
    }
}

File: sdks/csharp/src/Blok.Core/Types/NodeConfig.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Blok.Core.Types;

/// <summary>
/// NodeConfig represents node-specific configuration from the runner.
/// </summary>
public class NodeConfig
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("path")]
    public string Path { get; set; } = string.Empty;

    [JsonPropertyName("type")]
    public string Type { get; set; } = string.Empty;

    [JsonPropertyName("config")]
    public Dictionary<string, JsonElement> Config { get; set; } = new();

    /// <summary>
    /// Get a string config value with a default.
    /// </summary>
    public string GetConfigString(string key, string defaultValue = "")
    {
        if (Config.TryGetValue(key, out var element) && element.ValueKind == JsonValueKind.String)
        {
            return element.GetString() ?? defaultValue;
        }
        return defaultValue;
    }

    /// <summary>
    /// Get an integer config value with a default.
    /// </summary>
    public int GetConfigInt(string key, int defaultValue = 0)
    {
        if (Config.TryGetValue(key, out var element) && element.ValueKind == JsonValueKind.Number)
        {
            return element.TryGetInt32(out var result) ? result : defaultValue;
        }
        return defaultValue;
    }

    /// <summary>
    /// Get a boolean config value with a default.
    /// </summary>
    public bool GetConfigBool(string key, bool defaultValue = false)
    {
        if (Config.TryGetValue(key, out var element))
        {
            if (element.ValueKind == JsonValueKind.True) return true;
            if (element.ValueKind == JsonValueKind.False) return false;
        }
        return defaultValue;
    }
}
